### PR TITLE
Add probe ingress for ingress nginx

### LINKED ISSUE #1869: [2] Add probing endpoint for wc
**Is your feature request related to a problem? Please describe.**

To make sure platform administrators can use even if the application developers add a wildcard ingress object, we'll need to add an ingress for that purpose.

**Describe the solution you'd like**

Create an ingress object in wc that could be used as a probing endpoint.
Add documentation about it in the [public docs](https://github.com/elastisys/compliantkubernetes)

**Definition of done:**
When could this feature be considered done.

- There's an ingress object in wc that can be used for probing purposes even if there exists a wildcard ingress
- The public documentation has been updated to inform application developers about this

### PR DESCRIPTION
**Pull request by lunkan93**
> [!warning]
> **This is public repository, ensure not to disclose:**
> - [x] personal data beyond what is necessary for interacting with this pull request
> - [x] business confidential information, such as customer names

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [x] kind/feature       
- [ ] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change  
- [ ] kind/dev-change    
- [ ] kind/security      
- [ ] [kind/adr]()       

### What does this PR do / why do we need this PR?

Adds an ingress in WC that can be used for probing purposes even if there's a wildcard ingress that points to a service that does not respond with code 200.

- Fixes #1869 

#### Additional information to reviewers
I would not say I'm super happy about the naming here, suggestions very much welcomed :smile: 

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [x] The change is transparent
  - [ ] The change is disruptive
  - [ ] The change requires no migration steps
  - [ ] The change requires migration steps
- Metrics checks:
  - [ ] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [x] The change does not cause any dropped packages in the `NetworkPolicy Dashboard`
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy
- Bug checks:
  - [ ] The bug fix is covered by regression tests

### CODE REVIEW

File: helmfile/charts/ingress-nginx-probe-ingress/templates/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{{- $fullName := include "ingress-nginx-probe-ingress.fullname" . -}}
+{{- $svcPort := .Values.service.port -}}
+{{- $host := printf "%s.%s" "ingress-nginx" (required "Cluster domain required" .Values.ingress.clusterDomain) -}}
```
Comment (aarnq) [inline helmfile/charts/ingress-nginx-probe-ingress/templates/ingress.yaml:3]: This should be set via `ingressNginx.subdomain` as we do for the rest of our ingresses.

Preferably this should be put together in the helmfile values file instead, and then you should add placeholders for everything into the chart values file so we actually have all available options.

**Review by aarnq — APPROVED** [review]

File: helmfile/charts/ingress-nginx-probe-ingress/values.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+fullnameOverride: "ingress-nginx-controller-metrics"
```
Comment (OlleLarsson) [inline helmfile/charts/ingress-nginx-probe-ingress/values.yaml:1]: Isn't this fullnameoverride something that is generally not set in helm chart values, where the name is just derived nicely from the release name?
```suggestion
```

Comment (lunkan93) [reply helmfile/charts/ingress-nginx-probe-ingress/values.yaml:1]: Yeah, I could change it back and then just hard code this or set it from a separate value
```yaml
paths:
  - backend:
      service:
        name: "ingress-nginx-controller-metrics"
```

File: helmfile/values/ingress-nginx-probe-ingress.yaml.gotmpl [hunk 3]
```diff
@@ -0,0 +1,7 @@
+ingress:
+  host: {{ .Values.ingressNginx.subDomain }}.{{ .Values.global.baseDomain }}
+  annotations:
+    cert-manager.io/cluster-issuer: {{ .Values.global.issuer }}
+  {{ if and .Values.externalTrafficPolicy.local .Values.externalTrafficPolicy.whitelistRange.wcProbeIngress }}
+    nginx.ingress.kubernetes.io/whitelist-source-range: {{ .Values.externalTrafficPolicy.whitelistRange.wcProbeIngress }}
```
Comment (Pavan-Gunda) [inline helmfile/values/ingress-nginx-probe-ingress.yaml.gotmpl:6]:
```suggestion
    nginx.ingress.kubernetes.io/whitelist-source-range: {{ .Values.externalTrafficPolicy.whitelistRange.global }}
```

Shouldn't it be this ?

File: helmfile/charts/ingress-nginx-probe-ingress/values.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+nameOverride: ""
+
+service:
+  name: "ingress-nginx-controller-metrics"
+  port: 10254
+
+ingress:
```
Comment (Xartos) [inline helmfile/charts/ingress-nginx-probe-ingress/values.yaml:7]: Might want to have some default for the `host` and `annotation` variable here as well

```suggestion
ingress:
  host: ingress.example.com
  annotations: {}
```

**Review by Xartos — APPROVED** [review]

**Pull request merged**